From: NetHunter Team <nethunter@kali.org>
Date: Tue, 01 Jul 2026
Subject: [PATCH 1/3] Add HID gadget support for NetHunter BadUSB attacks

Enable USB HID gadget mode for keyboard/mouse emulation attacks.
Required for NetHunter HID attacks feature.

---
 drivers/usb/gadget/Kconfig | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/drivers/usb/gadget/Kconfig b/drivers/usb/gadget/Kconfig
index abc1234..def5678 100644
--- a/drivers/usb/gadget/Kconfig
+++ b/drivers/usb/gadget/Kconfig
@@ -1,6 +1,18 @@
 # USB Gadget support for NetHunter

+config USB_CONFIGFS_F_HID
+	bool "HID gadget"
+	depends on USB_CONFIGFS
+	select USB_LIBCOMPOSITE
+	select HID
+	help
+	  This option adds HID gadget support to USB configfs.
+	  With this you can emulate a USB keyboard and mouse,
+	  which is required for NetHunter HID attacks.
+
+	  To compile this as a module, choose M here: the
+	  module will be called usb_f_hid.
+
 config USB_ZERO
 	tristate "Gadget Zero (DEVELOPMENT)"
 	depends on USB_GADGET
